utils/unminify.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Copyright 2017 The AMP Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview
+ * Convert's stacktrace line, column number and file references from minified
+ * to unminified.
+ * @type sourceMapConsumerCache = {
+ *  key : url to JS file
+ *  sourceMapConsumer : corresponding sourceMapConsumer.
+ * }
+ * @type requestCache = {
+ *  key : url of sourceMap requested
+ *  promise : promise that resolves to sourceMapConsumer
+ * }
```

Comment:
Move type declarations to where the variables are declared:
```js
/** @type {!Object<typeOfKey, typeOfValue>} */
const sourceMapConsumerCache = new Map();
```
https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler